tools/clipper.js

```diff
@@ -1,31 +1,126 @@
 /* eslint-disable no-console */
 const { nanoid } = require('nanoid');
-const { io } = require('../index.js');
+const { exec } = require('child_process');
+const os = require('os');
+const archiver = require('archiver');
+const { createClient } = require('webdav');
+const fs = require('fs');
+
+// Local files
+const config = require('../config.json');
 
 const idRegex = /(\?v=|be\/).{11}/g;
 
-function clipRequest([videoType, videoLink, timestamps, fileName, fileExt]) {
-	const id = videoLink.match(idRegex)[0].substring(3);
+const webdavClient = createClient(
+	config.webdav.url,
+	{
+		username: config.webdav.username,
+		password: config.webdav.password,
+		maxBodyLength: 100000000,
+		maxContentLength: 100000000,
+	},
+);
 
-	io.emit('request', {
-		internalId: `${id}_${nanoid()}`,
-		videoType,
-		videoLink,
-		timestamps,
-		fileName,
-		fileExt,
-	});
-}
-exports.clipRequest = clipRequest;
+async function clipRequest([videoType, videoLink, timestamps, projectName, fileExt, extraArgs]) {
+	if (!fs.existsSync('../download')) {
+		fs.mkdirSync('../download');
+	}
+	const internalId = `${videoLink.match(idRegex)[0].substring(3)}_${nanoid()}`;
+	let doNotStitch = false;
+	let rescaleVideo = false;
+	if (extraArgs[0].value === 'Do Not Stitch Clips' || extraArgs[1].value === 'Do Not Stitch Clips') {
+		doNotStitch = true;
+	}
+	if (extraArgs[0].value === 'Rescale Video' || extraArgs[1].value === 'Rescale Video') {
+		rescaleVideo = true;
+	}
```

Comment:
```suggestion
        if(extraArgs.length > 0) {
	    if (extraArgs[0].value === 'Do Not Stitch Clips' || (extraArgs[1] && extraArgs[1].value === 'Do Not Stitch Clips')) {
		  doNotStitch = true;
	     }
	     if (extraArgs[0].value === 'Rescale Video' || (extraArgs[1] && extraArgs[1].value === 'Rescale Video')) {
		  rescaleVideo = true;
	      }
	  }
```
Proper null checking, causes errors atm if the array is empty/null and it can't access the properties/items. Make sure to run this through eslint if copy pasting.